### PR TITLE
nixos boot doc: document boot.debug1devices

### PR DESCRIPTION
**Pull request by wmertens**
This hopefully prevents other people from encountering #62049

### CODE REVIEW

File: nixos/doc/manual/administration/boot-problems.xml [hunk 1]
```diff
@@ -38,6 +38,17 @@
      </para>
     </listitem>
    </varlistentry>
+   <varlistentry>
+    <term>
+     <literal>boot.debug1devices</literal>
+    </term>
+    <listitem>
+     <para>
+      Like <literal>boot.debug1</literal>, but with some more devices loaded.
```
Comment (Mic92) [inline nixos/doc/manual/administration/boot-problems.xml]:
```suggestion
      Like <literal>boot.debug1</literal>, but runs stage1 until kernel modules are loaded and device nodes are created.
```

**Comment by jakobrs** [comment]
Maybe `boot.debug1mounts` should be documented as well?

**Comment by wmertens** [comment]
@jakobrs so what would you propose for the text

```
      Like <literal>boot.debug1devices</literal>, but continues stage1 until <literal>neededForBoot</literal> file systems are mounted.
      This may help with e.g. [making the keyboard work].
```

Does debug1mounts go further than debug1devices?

**Comment by jakobrs** [comment]
That's pretty much it, yes. It's basically just a variant of `boot.debug1devices` that fails only after all `neededForBoot` file systems are mounted, which may or may not be beneficial depending on what you're trying to do.

There are some technical details (see [here](https://github.com/NixOS/nixpkgs/blob/48fc7fce1ca8aef3d7ceb33716e71c901438f0d2/nixos/modules/system/boot/stage-1-init.sh#L546-L580)), but those don't really matter. `boot.debug1mounts` fails right before stage 2 is launched, so it can be useful if you want to modify stuff inside the initrd with the system being "as usable as possible". And as a bonus, the file system is mounted for you.

It's also worth pointing out that if you `exit` from the shell launched by `boot.debug1mounts`, `boot.debug1devices`, `boot.debug1`, `debug1` or `boot.shell_on_fail`, the boot will proceed normally, from where that option was handled. So you can, for example, use `boot.debug1mounts` to move system files before starting the rest of the system. And then you can just `exit` and the boot process will proceed as usual, provided you haven't broken anything. This does not apply if you choose the "start an interactive shell *having pid 1*" option, because the new shell is `exec`ed

**Comment by wmertens** [comment]
@jakobrs you seem to know it in a lot more detail than me :) Would you mind updating the documentation for these options? If not, I'll try to copy what you wrote.

**Comment by wmertens** [comment]
merged, sorry for missing it

**Pull request merged**